### PR TITLE
Display more verbose error with instructions when user denies access to the addon

### PR DESCRIPTION
**Pull request by itsjackgardner**
Screen Shot 2021-07-26 at 3 20 48 pm

### CODE REVIEW

File: src/bitbucket/descriptor.ts [hunk 1]
```diff
@@ -1,80 +1,93 @@
...
+
+  const addonName = `Landkid Queue${
+    process.env.LANDKID_DEPLOYMENT !== 'prod'
+      ? ` (${process.env.LANDKID_DEPLOYMENT || 'local'})`
+      : ''
+  }`;
+
+  const params = new URLSearchParams();
+  params.append('appName', appName);
+  const appNameQueryString = params.toString();
```
Comment (itsjackgardner) [inline src/bitbucket/descriptor.ts:21]: Passing the app name in as a query string param was the only way I could figure out to access it from the addon.

File: src/static/bitbucket/components/App.tsx [hunk 2]
```diff
@@ -22,18 +31,22 @@ type AppState = {
   bannerMessage: BannerMessage | null;
 };
 
-export class App extends React.Component {
+const qs = new URLSearchParams(window.location.search);
+
+export class App extends React.Component<{}, AppState> {
+  appName = qs.get('appName') || 'Landkid';
```
Comment (itsjackgardner) [inline src/static/bitbucket/components/App.tsx:37]: Maybe there's a better default (i.e. point to the app name displayed above the addon)?

File: src/bitbucket/descriptor.ts (hunk 1 above)
Comment (lukebatchelor) [reply src/bitbucket/descriptor.ts:21]: Hacky solution, but could we bake it into `src/static/bitbucket/components/App.tsx` directly? Would need to do something like adding it as an env var at build time though. Which is super doable, but is it more or less annoying that getting the addon refreshed? 🤔

File: src/bitbucket/descriptor.ts [hunk 3]
```diff
@@ -1,80 +1,93 @@
...
+    // pullrequest:write requires repository:write so we don't need to define that as well
+    scopes: ['account', 'pullrequest:write', 'pipeline:write'],
+    authentication: {
+      type: 'jwt',
     },
-    webPanels: [
-      {
-        weight: 100,
-        tooltip: {
-          value: 'Packages to be released in this PR',
```
Comment (lukebatchelor) [inline src/bitbucket/descriptor.ts:43]: Ha!

File: src/static/bitbucket/components/App.tsx (hunk 2 above)
Comment (lukebatchelor) [reply src/static/bitbucket/components/App.tsx:37]: I think this is fine tbh.

**Comment by lukebatchelor** [comment]
Sweet, lgtm!

**Review by lukebatchelor — APPROVED** [review]

**Pull request merged**